// node-cache-lite/dist/cache/cacheStore.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CacheStore = void 0;
const ttlManager_1 = require("./ttlManager");
const now_1 = require("../utils/now");
class CacheStore {
    constructor(maxSize, enableStats = false) {
        this.store = new Map();
        this.ttlManager = new ttlManager_1.TTLManager();
        this.stats = {
            hits: 0,
            misses: 0,
            sets: 0,
            deletes: 0,
            evictions: 0,
            expirations: 0,
        };
        this.enableStats = false;
        this.maxSize = maxSize;
        this.enableStats = enableStats;
    }
    get(key) {
        const entry = this.store.get(key);
        if (!entry) {
            if (this.enableStats)
                this.stats.misses++;
            return undefined;
        }
        if (this.ttlManager.isExpired(key)) {
            this.delete(key);
            if (this.enableStats)
                this.stats.expirations++;
            if (this.enableStats)
                this.stats.misses++;
            return undefined;
        }
        if (this.enableStats)
            this.stats.hits++;
        return entry.value;
    }
    set(key, value, ttl) {
        if (this.maxSize && this.store.size >= this.maxSize && !this.store.has(key)) {
            const firstKey = this.store.keys().next().value;
            if (this.enableStats)
                this.stats.evictions++;
            this.delete(firstKey);
        }
        this.store.set(key, {
            value: value,
            expiresAt: ttl ? (0, now_1.now)() + ttl : null,
            createdAt: (0, now_1.now)(),
        });
        this.ttlManager.set(key, ttl);
        if (this.enableStats)
            this.stats.sets++;
    }
    delete(key) {
        this.ttlManager.delete(key);
        const deleted = this.store.delete(key);
        if (deleted && this.enableStats)
            this.stats.deletes++;
        return deleted;
    }
    clear() {
        this.store.clear();
        this.ttlManager.clear();
    }
    has(key) {
        if (this.ttlManager.isExpired(key)) {
            this.delete(key);
            return false;
        }
        return this.store.has(key);
    }
    size() {
        return this.store.size;
    }
    cleanupExpired() {
        const keys = Array.from(this.store.keys());
        const expiredKeys = this.ttlManager.cleanupExpired(keys);
        for (const key of expiredKeys) {
            this.store.delete(key);
        }
    }
    getAll() {
        return new Map(this.store);
    }
    getOrSet(key, loader, ttl) {
        const cached = this.get(key);
        if (cached !== undefined) {
            return cached;
        }
        const result = loader();
        if (result instanceof Promise) {
            return result.then(value => {
                this.set(key, value, ttl);
                return value;
            });
        }
        else {
            this.set(key, result, ttl);
            return result;
        }
    }
    entries() {
        return Array.from(this.store.entries()).map(([key, entry]) => [key, entry.value]);
    }
    keys() {
        return Array.from(this.store.keys());
    }
    values() {
        return Array.from(this.store.values()).map(entry => entry.value);
    }
    setMultiple(entries) {
        for (const [key, value, ttl] of entries) {
            this.set(key, value, ttl);
        }
    }
    deleteMultiple(keys) {
        let count = 0;
        for (const key of keys) {
            if (this.delete(key)) {
                count++;
            }
        }
        return count;
    }
    getMultiple(keys) {
        return keys.map(key => this.get(key)).filter(v => v !== undefined);
    }
    getStats() {
        if (!this.enableStats) {
            return null;
        }
        const total = this.stats.hits + this.stats.misses;
        const hitRate = total === 0 ? 0 : (this.stats.hits / total) * 100;
        return {
            ...this.stats,
            hitRate,
        };
    }
}
exports.CacheStore = CacheStore;
