// node-cache-lite/dist/cache/createCache.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createCache = createCache;
const cacheStore_1 = require("./cacheStore");
function createCache(options) {
    const { ttl = undefined, maxSize = undefined, enableStats = false } = options || {};
    const cacheStore = new cacheStore_1.CacheStore(maxSize, enableStats);
    return {
        get(key) {
            return cacheStore.get(key);
        },
        set(key, value, ttl) {
            cacheStore.set(key, value, ttl);
        },
        delete(key) {
            return cacheStore.delete(key);
        },
        clear() {
            cacheStore.clear();
        },
        has(key) {
            return cacheStore.has(key);
        },
        size() {
            return cacheStore.size();
        },
        getOrSet(key, loader, ttl) {
            return cacheStore.getOrSet(key, loader, ttl);
        },
        entries() {
            return cacheStore.entries();
        },
        keys() {
            return cacheStore.keys();
        },
        values() {
            return cacheStore.values();
        },
        setMultiple(entries) {
            cacheStore.setMultiple(entries);
        },
        deleteMultiple(keys) {
            return cacheStore.deleteMultiple(keys);
        },
        getMultiple(keys) {
            return cacheStore.getMultiple(keys);
        },
        getStats() {
            return cacheStore.getStats();
        },
    };
}
