// node-cache-lite/dist/cache/ttlManager.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TTLManager = void 0;
const isExpired_1 = require("../utils/isExpired");
class TTLManager {
    constructor() {
        this.expiryMap = new Map();
    }
    set(key, ttl) {
        if (ttl === undefined || ttl === null) {
            this.expiryMap.set(key, null);
        }
        else {
            const expiresAt = Date.now() + ttl;
            this.expiryMap.set(key, expiresAt);
        }
    }
    isExpired(key) {
        const expiresAt = this.expiryMap.get(key);
        if (expiresAt === undefined) {
            return false;
        }
        return (0, isExpired_1.isExpired)(expiresAt);
    }
    delete(key) {
        this.expiryMap.delete(key);
    }
    clear() {
        this.expiryMap.clear();
    }
    getExpiresAt(key) {
        return this.expiryMap.get(key);
    }
    cleanupExpired(keys) {
        const expiredKeys = [];
        for (const key of keys) {
            if (this.isExpired(key)) {
                expiredKeys.push(key);
                this.delete(key);
            }
        }
        return expiredKeys;
    }
}
exports.TTLManager = TTLManager;
