// node-cache-lite/dist/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.unhashKey = exports.hashKey = exports.now = exports.isExpired = exports.CacheError = exports.TTLManager = exports.CacheStore = exports.createCache = void 0;
var createCache_1 = require("./cache/createCache");
Object.defineProperty(exports, "createCache", { enumerable: true, get: function () { return createCache_1.createCache; } });
var cacheStore_1 = require("./cache/cacheStore");
Object.defineProperty(exports, "CacheStore", { enumerable: true, get: function () { return cacheStore_1.CacheStore; } });
var ttlManager_1 = require("./cache/ttlManager");
Object.defineProperty(exports, "TTLManager", { enumerable: true, get: function () { return ttlManager_1.TTLManager; } });
var CacheError_1 = require("./errors/CacheError");
Object.defineProperty(exports, "CacheError", { enumerable: true, get: function () { return CacheError_1.CacheError; } });
var isExpired_1 = require("./utils/isExpired");
Object.defineProperty(exports, "isExpired", { enumerable: true, get: function () { return isExpired_1.isExpired; } });
var now_1 = require("./utils/now");
Object.defineProperty(exports, "now", { enumerable: true, get: function () { return now_1.now; } });
var hashKey_1 = require("./utils/hashKey");
Object.defineProperty(exports, "hashKey", { enumerable: true, get: function () { return hashKey_1.hashKey; } });
Object.defineProperty(exports, "unhashKey", { enumerable: true, get: function () { return hashKey_1.unhashKey; } });
